### PR TITLE
V4 bmfont improved api

### PR DESCRIPTION
**Pull request by rh101**
Merged the code changes that were added in V3 regarding the BM Font API.  This adds support for embedding BMFont atlases into a larger sprite atlas, which also supports rotated images.

### CODE REVIEW

**Comment by rh101** [comment]
@minggo  This code is based on V3 PRs #20300 and #20309 as I didn't realise they were never merged into V4 as well.

File: tests/cpp-tests/Classes/LabelTest/LabelTestNew.cpp [hunk 1]
```diff
@@ -933,6 +935,57 @@ std::string LabelFNTBounds::subtitle() const
...
+{
+    return "Using 2 .fnt definitions that share the same texture atlas.";
+}
+
+LabelFNTMultiFontAtlasWithRotation::LabelFNTMultiFontAtlasWithRotation()
+{
+    auto s = Director::getInstance()->getWinSize();
+
+    auto spriteCache = SpriteFrameCache::getInstance();
+    spriteCache->addSpriteFramesWithFile("fonts/bmfont-rotated-test.plist");
```
Comment (PatriceJiang) [inline tests/cpp-tests/Classes/LabelTest/LabelTestNew.cpp:966]: Font resources for tests are missing

Comment (rh101) [reply tests/cpp-tests/Classes/LabelTest/LabelTestNew.cpp:966]: I just realised that too, and updated them as you typed the comment. Sorry about that!

Comment (PatriceJiang) [reply tests/cpp-tests/Classes/LabelTest/LabelTestNew.cpp:966]: It’s ok, I didn't realise the "Refresh" notice.

**Review by PatriceJiang — APPROVED** [review]

**Pull request merged**